### PR TITLE
v2 docs for ScaledObject

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubali@redhat.com>

### CODE REVIEW

File: content/docs/2.0/concepts/scaling-deployments.md [hunk 1]
```diff
@@ -57,13 +59,19 @@ You can find all supported triggers [here](/scalers).
 ### Details
 ```yaml
   scaleTargetRef:
-    deploymentName: {deployment-name} # must be in the same namespace
-    containerName: {container-name}  #Optional. Default: deployment.spec.template.spec.containers[0]
+    apiVersion:    {api-version-of-target-resource}  # Optional. Default: apps/v1
+    kind:          {kind-of-target-resource}         # Optional. Default: Deployment
+    name:          {name-of-target-resource}         # Must be in the same namespace as the ScaledObject
```
Comment (tomkerkhove) [inline content/docs/2.0/concepts/scaling-deployments.md:64]:
```suggestion
    name:          {name-of-target-resource}         # Mandatory. Must be in the same namespace as the ScaledObject
```

**Comment by zroubalik** [comment]
This PR is still in WIP, I'll let you know once it is ready for review, thanks!

**Comment by zroubalik** [comment]
Ready for a review.

File: content/docs/2.0/concepts/scaling-deployments.md [hunk 2]
```diff
@@ -32,38 +40,52 @@ metadata:
...
+    containerName: {container-name}                  # Optional. Default: .spec.template.spec.containers[0]
+  pollingInterval: 30                                # Optional. Default: 30 seconds
+  cooldownPeriod:  300                               # Optional. Default: 300 seconds
+  minReplicaCount: 0                                 # Optional. Default: 0
+  maxReplicaCount: 100                               # Optional. Default: 100
+  advanced:                                          # Optional. Section to specify advanced options.
+    horizontalPodAutoscalerConfig:                   # Optional. If not set, KEDA won't scale based on resource utilization
       resourceMetrics:
-      - name: cpu/memory # Name of the resource to be targeted
+      - name: cpu/memory                             # Name of the resource to be targeted
```
Comment (tomkerkhove) [inline content/docs/2.0/concepts/scaling-deployments.md:54]: What about this?
```suggestion
      - name: cpu/memory                             # Name of the metric to scale on
```

File: content/docs/2.0/concepts/scaling-deployments.md [hunk 3]
```diff
@@ -32,38 +40,52 @@ metadata:
...
 
-The name of the deployment this scaledObject is for. This is the deployment KEDA will scale up and setup an HPA for based on the triggers defined in `triggers:`. Make sure to include the deployment name in the label as well, otherwise the metrics provider will not be able to query the metrics for the scaled object and 1-n scale will be broken.
+The reference to the resource this ScaledObject is configured for. This is the resource KEDA will scale up/down and setup an HPA for, based on the triggers defined in `triggers:`. 
+
+To scale Kubernetes Deployments only `name` is needed to be specified, if one wants to scale a different resource such as StatefulSet or  Custom Resource (that defines `/scale` subresource), appropriate `apiVersion` (following standard Kubernetes convetion, ie. `{api}/{version}`) and `kind` need to be specfied.
 
-**Assumptions:** `deploymentName` is in the same namespace as the scaledObject
+`containerName` is a name of container in the target resource, from which KEDA should try to get environment properties holding secrets etc.
+
+**Assumptions:** Resource referenced by `name` (and `apiVersion`, `kind`) is in the same namespace as the scaledObject
```
Comment (tomkerkhove) [inline content/docs/2.0/concepts/scaling-deployments.md:88]: Should we put it like this or be explicit and say we only support scaling workloads in the same namespace as the SO?

Comment (zroubalik) [reply content/docs/2.0/concepts/scaling-deployments.md:88]: What do you propose by explicit?

File: content/docs/2.0/concepts/scaling-deployments.md (hunk 2 above)
Comment (zroubalik) [reply content/docs/2.0/concepts/scaling-deployments.md:54]: Good catch, makes sense, will fix that

**Pull request merged**